packages/timeline-state-resolver/src/integrations/casparCG/index.ts

```diff
@@ -849,83 +699,70 @@ export class CasparCGDevice extends DeviceWithState<State, DeviceOptionsCasparCG
+			if (!response) return
+
+			if (
+				response.responseCode < 300 && // TODO - maybe we accept every code except 404?
+				response.command.match(/Loadbg|Play|Load|Clear|Stop|Resume/i) &&
+				'channel' in cmd.params &&
+				cmd.params.channel !== undefined &&
+				'layer' in cmd.params &&
+				cmd.params.layer !== undefined
+			) {
+				const currentState = this.getState(time)
+				if (currentState) {
+					const currentCasparState = currentState.state
+
+					const trackedState = this._ccgState.getState()
+
+					const channel = currentCasparState.channels[cmd.params.channel]
+					if (channel) {
+						if (!trackedState.channels[cmd.params.channel]) {
+							trackedState.channels[cmd.params.channel] = {
+								channelNo: channel.channelNo,
+								fps: channel.fps || 0,
+								videoMode: channel.videoMode || null,
+								layers: {},
 							}
-							// Copy the tracked from current state:
-							trackedState.channels[resCommand.channel].layers[resCommand.layer] = channel.layers[resCommand.layer]
-							this._ccgState.setState(trackedState)
 						}
+						// Copy the tracked from current state:
+						trackedState.channels[cmd.params.channel].layers[cmd.params.layer] = channel.layers[cmd.params.layer]
+						this._ccgState.setState(trackedState)
 					}
 				}
+			}
 
-				if (this._queueOverflow) {
-					this._queueOverflow = false
-					this._connectionChanged()
-				}
-			})
-			.catch((error) => {
-				if (error?.response?.code === 504) {
-					if (!this._queueOverflow) {
-						this._queueOverflow = true
-						this._connectionChanged()
-					}
-				}
+			if (response.responseCode === 504 && !this._queueOverflow) {
+				this._queueOverflow = true
+				this._connectionChanged()
+			} else if (this._queueOverflow) {
+				this._queueOverflow = false
+				this._connectionChanged()
+			}
 
-				let errorString = ''
-				if (error?.response?.code === 404) {
-					errorString = `404: File not found`
-				}
+			if (response.responseCode >= 400) {
+				// this is an error code:
+				let errorString = `${response.responseCode} ${cmd.command} ${response.type}: ${response.type}`
 
-				if (!errorString) {
-					errorString = error?.response?.raw ? error.response.raw : error.toString()
+				if (Object.keys(cmd.params).length) {
+					errorString += ' ' + JSON.stringify(cmd.params)
 				}
 
-				if (cmd.name) {
-					errorString += ` ${cmd.name} `
-				}
-				if (cmd['_objectParams'] && !_.isEmpty(cmd['_objectParams'])) {
-					errorString += ', params: ' + JSON.stringify(cmd['_objectParams'])
-				} else if (cmd.payload && !_.isEmpty(cmd.payload)) {
-					errorString += ', payload: ' + JSON.stringify(cmd.payload)
-				}
 				this.emit('commandError', new Error(errorString), cwc)
-				if (cmd.name === 'ScheduleSetCommand') {
-					// delete this._queue[cmd.getParam('command').token]
-					delete this._queue[cmd.token]
-				}
-			})
+			}
+		} catch (e) {
+			// This shouldn't really happen
```

Comment:
Would be really nice if we didn't have comments like this in the code...
Comments should provide extra insight. If they don't, then they should be deleted